topdown/http.go

```diff
@@ -229,6 +231,41 @@ func canonicalizeHeaders(headers map[string]interface{}) map[string]interface{}
 	return canonicalized
 }
 
+// useSocket examines the url for "unix://" and returns a *http.Transport with
+// a DialContext that opens a socket (specified in the http call).
+// The url is expected to contain socket=/path/to/socket (url encoded)
+// Ex. "unix:localhost/end/point?socket=%2Ftmp%2Fhttp.sock"
+func useSocket(rawURL string, tlsConfig *tls.Config) (bool, string, *http.Transport) {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return false, rawURL, nil
+	}
+
+	if u.Scheme == "unix" {
+		// Get the path to the socket
+		if u.RawQuery != "" {
+			v, err := url.ParseQuery(u.RawQuery)
+			if err != nil {
+				return false, rawURL, nil
+			}
+
+			tr := http.Transport{
```

Comment:
❓ We're creating new `http.Transport` structs here. Unless something has changed, these, having their golang struct zero values, are not good defaults. In Golang 1.13, [`Clone()`](https://pkg.go.dev/net/http#Transport.Clone) was introduced, and I wonder if `http.DefaultTransport.Clone()` wouldn't be the right way to initialize `tr`.... What do you think?

Reply:
@srenatus Thanks for the great feedback! 👍 I updated the PR.  Clone() rocks!:-D